examples/training_without_local_storage.ipynb

```diff
@@ -0,0 +1,1005 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorboard.notebook\n",
+    "\n",
+    "%tensorboard --logdir s3://{s3_bucket_name}/{bucket_prefix}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Congratulations! You trained your first model without relying on a persistent local disk!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Putting it all together\n",
+    "\n",
+    "In this tutorial, we walked through storing and loading checkpoints with the cloud, converting existing datasets into a streaming format, and training a model using both of these features.\n",
+    "\n",
+    "Below is a complete example, showing everything we did in one cell. Feel free to use this as a reference for your own training workflows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import io\n",
+    "import os\n",
+    "import struct\n",
+    "import shutil\n",
+    "import time\n",
+    "from typing import Dict, cast, Iterable, Tuple\n",
+    "\n",
+    "import torch\n",
+    "from torch.optim import SGD\n",
+    "import torch.utils.data\n",
+    "from torchvision.datasets import MNIST\n",
+    "from torchvision.transforms import ToTensor\n",
+    "from tqdm.auto import tqdm\n",
+    "\n",
+    "from composer import Trainer\n",
+    "from composer.models import mnist_model\n",
+    "from composer.utils.reproducibility import seed_all\n",
+    "from composer.utils.object_store import S3ObjectStore\n",
+    "from composer.loggers import ObjectStoreLogger, TensorboardLogger\n",
+    "from composer.datasets.streaming import StreamingDatasetWriter, StreamingDataset\n",
+    "\n",
+    "\n",
+    "###############\n",
+    "# Configuration\n",
+    "###############\n",
+    "\n",
+    "# # The S3 bucket to use\n",
+    "# s3_bucket_name = 'my-bucket'\n",
+    "\n",
+    "# # Prefix all objects in the bucket with key,\n",
+    "# # allowing the bucket can be shared across training runs\n",
+    "# bucket_prefix = 'composer-diskless-training-tutorial'\n",
+    "\n",
+    "# # If necessary, uncomment the following lines to set AWS credentials\n",
+    "# # Do NOT include quotes\n",
+    "# %env AWS_ACCESS_KEY_ID ***\n",
+    "# %env AWS_SECRET_ACCESS_KEY ***\n",
+    "# %env AWS_DEFAULT_REGION us-west-2\n",
+    "\n",
+    "# # Also define local (temporary) folders that will be used for\n",
+    "# # staging datasets, checkpoints, and log files before uploading\n",
+    "# data_dir = '/tmp/data'\n",
+    "# tensorboard_log_dir = '/tmp/tb_logs'\n",
+    "# checkpoint_dir = '/tmp/checkpoints'\n",
```

Comment:
uncomment so it works when copy pasted?

Reply:
I decided to leave this commented out so it does not override the configuration that may have been set at the top of the file. But wanted to include it here again if someone happens to copy and paste this script.